engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
+      }
     }
   }
 
+  private boolean shouldLog()
+  {
+    if ((lastLog == -1) || (totalTupleCount - lastLog) >= ERROR_LOG_GAP) {
```

Comment:
The intent of the property is to log an error if at least x tuples have occurred in between and not log every xth error. The case that produced this fix was that every tuple was failing serialization because it was not json serializable and this will lead to computation for every tuple.